ssts/segmentation/segmentation_gmm.py

```diff
@@ -112,20 +112,19 @@ def transform(self, data, outliers=None):
         else:
             data = data.reshape(n, c)
 
-        if outliers is not None:
-            outliers = outliers.flatten()
-
-        data = SegmenterGMM.remove_outliers(data, outliers)
-
         if self.normalize:
             data = self.sst.transform(data) if self.zscale else SegmenterGMM.normalize_by_max(data)
 
         if self.embedding_dim is not None:
             data = self.pca.transform(data)
 
         labels = self.gmm.predict(data)
+        labels = np.reshape(labels, (h, w))
+        labels += 1 # all labels move up one
+        if outliers is not None:
+            labels *= (1 - outliers) # outliers map to label 0
```

Comment:
I solve the issue addressed above by first classifying all the pixels in the original image, and then mapping the outlier mask to zeros